scripts/zkai_native_seq32_attention_mlp_single_proof_gate.py

```
@@ -0,0 +1,641 @@
+    writer.writeheader()
+    writer.writerow(row)
+    return buffer.getvalue()
+
+
+def atomic_write_text(path: pathlib.Path, text: str) -> None:
+    root = ROOT.resolve()
+    target = pathlib.Path(os.path.abspath(path if path.is_absolute() else ROOT / path))
+    try:
+        target.relative_to(root)
+    except ValueError as err:
+        raise NativeSeq32AttentionMlpSingleProofGateError("output path escapes repo root") from err
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)